src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs

```diff
@@ -31,40 +25,26 @@ partial class Shape
 		protected Shape() : base("svg", isSvg: true)
 		{
 			SvgChildren = new UIElementCollection(this);
-			SvgChildren.CollectionChanged += OnSvgChildrenChanged;
-
-			OnStretchUpdatedPartial();
-		}
-
-		protected void InitCommonShapeProperties() // Should be called from base class constructor
-		{
-			// Initialize
-			OnFillUpdatedPartial(); // Required to properly update the HitTest
-									// Don't OnStrokeUpdatedPartial(); => Stroke is still at its default value in the ctor, and it will always results to ResetStyle("stroke")
-									// Don't OnStrokeThicknessUpdatedPartial(); => The default value is set in Uno.UI.css
 		}
 
 		protected abstract SvgElement GetMainSvgElement();
 
-		private static readonly NotifyCollectionChangedEventHandler OnSvgChildrenChanged = (object sender, NotifyCollectionChangedEventArgs args) =>
-		{
-			if (sender is UIElementCollection children && children.Owner is Shape shape)
-			{
-				shape.OnChildrenChanged();
-			}
-		};
-
-		protected virtual void OnChildrenChanged()
+		private protected void UpdateRender()
 		{
+			UpdateFill();
+			UpdateStroke();
+			UpdateStrokeThickness();
+			UpdateStrokeDashArray();
```

Comment:
I think Stroke should affect measure as well?

Consider a shape with StrokeThickness set to some number, and then a stroke changing from null to non-null or vice-versa. The "effective" stroke thickness would have changed with this stroke change, if I'm not mistaken?

Reply:
Indeed, so we need to update the `Stroke` DP to require to invalidate the measure https://github.com/unoplatform/uno/pull/13220/files/3d2af3d34765c2fc718cf40b9514ccb80cee9435#diff-1cfc420fdae934149bc7328e7d89a2143a73dd3d898d322d15dad1002c8980cbR115 (for this particular case we could proabbly require an `InvalidateMeasure` only when we go from / to `null` to / from `not null` instead of `AffectsMeasure` flag)

Reply:
Also we could update the `StrokeThickness` property to invalidate the measure only if the `Stroke is not null`.